src/universal/modules/invoice/containers/InvoiceRoot.js

```diff
@@ -0,0 +1,49 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {TransitionGroup} from 'react-transition-group';
+import AnimatedFade from 'universal/components/AnimatedFade';
+import ErrorComponent from 'universal/components/ErrorComponent/ErrorComponent';
+import LoadingComponent from 'universal/components/LoadingComponent/LoadingComponent';
+import QueryRenderer from 'universal/components/QueryRenderer/QueryRenderer';
+import withAtmosphere from 'universal/decorators/withAtmosphere/withAtmosphere';
+import Invoice from 'universal/modules/invoice/components/Invoice/Invoice';
+
+const query = graphql`
+  query InvoiceRootQuery($invoiceId: ID!) {
+    viewer {
+      ...Invoice_viewer
+    }
+  }
+`;
+
+const InvoiceRoot = ({atmosphere, match: {params: {invoiceId}}}) => {
+  return (
+    <QueryRenderer
+      environment={atmosphere}
+      query={query}
+      variables={{invoiceId}}
+      render={({error, props: queryProps}) => {
+        return (
```

Comment:
tell me about it! that's only of my beefs with relay modern, if you think of a way to abstract it, i'd love to see it!

Reply:
Alright, deal!

We could start by naming this render function:

```js
const queryRender = ({ error, props }) => (
  <TransitionGroup {/* */}>
    {/* */}
  </TransitionGroup>
);
```

I actually think the rest of my legibility concern could be addressed by indentation; instead of:

```js
{queryProps && <AnimatedFade key="1">
  <Invoice viewer={queryProps.viewer} />
  </AnimatedFade>}
```

perhaps:

```js
{queryProps &&
  <AnimatedFade key="1">
    <Invoice viewer={queryProps.viewer} />
  </AnimatedFade>
}
```

Reply:
we could name it, but we'd have to still declare it in the same scope because we need the props passed in from the parent scope, so that'd just make it longer, unless you wanna curry those into the renderProps?